salt/modules/disk.py

```diff
@@ -481,12 +481,13 @@ def fstype(device):
                 fs_type = df_out[2]
                 if fs_type:
                     return fs_type
-        else:
-            df_out = __salt__['cmd.run']('df -T {0}'.format(device)).splitlines()
-            if len(df_out) > 1:
-                fs_type = df_out[1]
-                if fs_type:
-                    return fs_type
```

Comment:
I generally operate on the assumption that less code is better so I removed the extra branch. 

Since `file -s` works in both cases it seems simpler to me to use it in both cases, but I can update the PR to put the branch back in if you would prefer.